services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java

```diff
@@ -1307,15 +1308,12 @@ public void onScanStart() {
 
     @Override
     public Long[] getScannablePools() {
-        List<DataCenterVO> zones = _dcDao.listEnabledZones();
-
-        Long[] dcIdList = new Long[zones.size()];
-        int i = 0;
-        for (DataCenterVO dc : zones) {
-            dcIdList[i++] = dc.getId();
+        List<DataCenterVO> zones = _dcDao.listEnabledNonEdgeZones();
+        List<Long> zoneIds = zones.stream().map(DataCenterVO::getId).collect(Collectors.toList());
```

Comment:
@nvazquez since listEnabledNonEdgeZones uses `GenericDaoBase::listBy` it won't return null. It will always return a list and an empty list when no zones found.

Reply:
@nvazquez I've refactored the code a bit. Can you please check?

Reply:
Thanks, looks good